crt_portal/tms/admin.py

```diff
@@ -1,10 +1,39 @@
-from cts_forms.admin import ReadOnlyModelAdmin
+import logging
+from cts_forms.admin import ReadOnlyModelAdmin, csv, Echo, iter_queryset, format_export_message
 from django.contrib import admin
 from django.utils.html import format_html
 from django.urls import reverse
+from django.http import StreamingHttpResponse
 
 from .models import TMSEmail
 
+TMS_FIELDS = ['id', 'tms_id', 'subject', 'body', 'recipient', 'status', 'error_message', 'report_id', 'created_at', 'completed_at', 'purpose']
+
+logger = logging.getLogger(__name__)
+
+
+def _serialize_tms(data):
+    """Preserve headers while rendering TMS_FIELDS for inbound actions"""
+    if isinstance(data, TMSEmail):
+        return [getattr(data, field) for field in TMS_FIELDS]
+    return data
+
+
+def export_tms_as_csv(modeladmin, request, queryset):
+    """
+    Stream actions as csv
+    Log all use
+    """
+    writer = csv.writer(Echo(), quoting=csv.QUOTE_ALL)
+    iterator = iter_queryset(queryset, TMS_FIELDS)
+    response = StreamingHttpResponse((writer.writerow(_serialize_tms(tms)) for tms in iterator),
```

Comment:
Since the iteration is dependent on the TMS fields, if we pass some other (non-TMS Model) array in there and it gets returned by serialize_tms, isn't this for loop gonna break?

Reply:
This export function is specifically for tms, so I don't think we would use for other models.  Its basically a copy of "export_actions_as_csv" over in tms_forms/admin.py

Reply:
Gotcha! Thanks @billyhunt